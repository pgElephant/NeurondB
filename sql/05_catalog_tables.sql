-- Test catalog table creation
SELECT tablename FROM pg_tables 
WHERE tablename LIKE 'neurondb_%' 
ORDER BY tablename;

-- Test job queue table
INSERT INTO neurondb_job_queue (job_type, payload, tenant_id) 
VALUES ('test_job', '{"key": "value"}'::jsonb, 1);
SELECT job_type, tenant_id, status FROM neurondb_job_queue;

-- Test query metrics table
INSERT INTO neurondb_query_metrics (query_type, latency_ms, recall_at_k, ef_search)
VALUES ('knn_search', 25.5, 0.95, 64);
SELECT query_type, ef_search FROM neurondb_query_metrics;

-- Test embedding cache table
INSERT INTO neurondb_embedding_cache (cache_key, embedding, model_name)
VALUES ('test_key', '[1.0, 2.0, 3.0]'::vector, 'test_model');
SELECT cache_key, model_name FROM neurondb_embedding_cache;

-- Count all catalog tables
SELECT COUNT(*) AS catalog_table_count FROM pg_tables WHERE tablename LIKE 'neurondb_%';

