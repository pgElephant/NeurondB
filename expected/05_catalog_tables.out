-- Test catalog table creation
SELECT tablename FROM pg_tables 
WHERE tablename LIKE 'neurondb_%' 
ORDER BY tablename;
 tablename 
-----------
(0 rows)

-- Test job queue table
INSERT INTO neurondb_job_queue (job_type, payload, tenant_id) 
VALUES ('test_job', '{"key": "value"}'::jsonb, 1);
ERROR:  relation "neurondb_job_queue" does not exist
LINE 1: INSERT INTO neurondb_job_queue (job_type, payload, tenant_id...
                    ^
SELECT job_type, tenant_id, status FROM neurondb_job_queue;
ERROR:  relation "neurondb_job_queue" does not exist
LINE 1: SELECT job_type, tenant_id, status FROM neurondb_job_queue;
                                                ^
-- Test query metrics table
INSERT INTO neurondb_query_metrics (query_type, latency_ms, recall_at_k, ef_search)
VALUES ('knn_search', 25.5, 0.95, 64);
ERROR:  relation "neurondb_query_metrics" does not exist
LINE 1: INSERT INTO neurondb_query_metrics (query_type, latency_ms, ...
                    ^
SELECT query_type, ef_search FROM neurondb_query_metrics;
ERROR:  relation "neurondb_query_metrics" does not exist
LINE 1: SELECT query_type, ef_search FROM neurondb_query_metrics;
                                          ^
-- Test embedding cache table
INSERT INTO neurondb_embedding_cache (cache_key, embedding, model_name)
VALUES ('test_key', '[1.0, 2.0, 3.0]'::vector, 'test_model');
ERROR:  relation "neurondb_embedding_cache" does not exist
LINE 1: INSERT INTO neurondb_embedding_cache (cache_key, embedding, ...
                    ^
SELECT cache_key, model_name FROM neurondb_embedding_cache;
ERROR:  relation "neurondb_embedding_cache" does not exist
LINE 1: SELECT cache_key, model_name FROM neurondb_embedding_cache;
                                          ^
-- Count all catalog tables
SELECT COUNT(*) AS catalog_table_count FROM pg_tables WHERE tablename LIKE 'neurondb_%';
 catalog_table_count 
---------------------
                   0
(1 row)

